pkg/sync/status.go

```diff
@@ -2,110 +2,111 @@ package sync
+				}
+			} else {
+				log.Info(resourceIDToString(gk, name),
+					"status", strings.ToLower(e.ApplyEvent.Status.String()))
+			}
+		case event.StatusType:
+			statusCollector.updateStatus(e.StatusEvent.Identifier, e.StatusEvent)
+			if printStatus {
+				gk := e.StatusEvent.Identifier.GroupKind
+				name := e.StatusEvent.Identifier.Name
+				if e.StatusEvent.Error != nil {
+					errorMsg := fmt.Sprintf("%s status %s: %s\n", resourceIDToString(gk, name),
+						strings.ToLower(e.StatusEvent.PollResourceInfo.Status.String()), e.StatusEvent.Error.Error())
+					err = fmt.Errorf(errorMsg)
+					log.Error(err, "status error")
+				} else {
+					log.Info(resourceIDToString(gk, name),
+						"status", strings.ToLower(e.StatusEvent.PollResourceInfo.Status.String()))
+				}
+			}
+		}
 	}
-}
 
-func (engine *Engine) updateStatus(id string, results []common.ResourceSyncResult, err *console.ServiceErrorAttributes) error {
-	errs := make([]*console.ServiceErrorAttributes, 0)
-	if err != nil {
-		errs = append(errs, err)
+	for _, v := range statusCollector.latestStatus {
+		consoleAttr := fromSyncResult(v)
+		components = append(components, consoleAttr)
 	}
 
-	components, compErr := engine.collectComponents(id, results)
-	if err != nil {
-		errs = append(errs, errorAttributes("reconciliation", compErr))
+	if err := engine.updateStatus(id, components, errorAttributes("sync", err)); err != nil {
+		log.Error(err, "Failed to update service status, ignoring for now")
 	}
-	errs = lo.Filter(errs, func(e *console.ServiceErrorAttributes, ind int) bool { return e != nil })
 
-	return engine.client.UpdateComponents(id, components, errs)
+	return nil
 }
 
-func (engine *Engine) collectComponents(id string, results []common.ResourceSyncResult) ([]*console.ComponentAttributes, error) {
-	res := make([]*console.ComponentAttributes, 0)
-	seen := map[kube.ResourceKey]bool{}
-	liveObjs, err := engine.cache.GetManagedLiveObjs([]*unstructured.Unstructured{}, isManaged(id))
-	if err != nil {
-		return res, err
-	}
-
-	for _, syncResult := range results {
-		fmt.Printf("%+v\n", syncResult)
-		component := fromSyncResult(syncResult)
-		obj, ok := liveObjs[syncResult.ResourceKey]
-		fmt.Printf("%+v\n", obj)
-		if ok {
-			seen[syncResult.ResourceKey] = true
-			gvk := obj.GroupVersionKind()
-			component.State = toStatus(obj)
-			component.Version = gvk.Version
-			component.Group = gvk.Group
-			component.Kind = gvk.Kind
-			component.Namespace = obj.GetNamespace()
-		}
-		res = append(res, component)
-	}
-
-	for k, obj := range liveObjs {
-		if seen[k] {
-			continue
-		}
-		gvk := obj.GroupVersionKind()
-		res = append(res, &console.ComponentAttributes{
-			State:     toStatus(obj),
-			Group:     gvk.Group,
-			Kind:      gvk.Kind,
-			Version:   gvk.Version,
-			Namespace: obj.GetNamespace(),
-			Name:      obj.GetName(),
-			Synced:    true,
-		})
+func fromSyncResult(e event.StatusEvent) *console.ComponentAttributes {
+	return &console.ComponentAttributes{
```

Comment:
I think this is missing the `State` field (which i think you can get from the khealth submodule of cli-utils)

Reply:
I have been working with this